cli/central-client/src/main/java/org/ballerinalang/central/client/Utils.java

```diff
@@ -274,6 +280,23 @@ private static void writeAndHandleProgress(InputStream inputStream, FileOutputSt
         }
     }
 
+    private static void writeAndHandleProgressWithoutLogging(InputStream inputStream, FileOutputStream outputStream,
+                                                             String fullPkgName, LogFormatter logFormatter) {
+        int count;
+        byte[] buffer = new byte[1024];
+
+        try {
+            while ((count = inputStream.read(buffer)) > 0) {
+                outputStream.write(buffer, 0, count);
+            }
+        } catch (IOException e) {
+            PrintStream printStream = System.out;
+
+            printStream.println(logFormatter.formatLog(fullPkgName + "pulling the package from central failed"));
```

Comment:
At this level this should be an exception and can be handled at a higher level. Then the exception can be converted to a diagnostic at the Package resolution